File: src/LeetCodeCS/InsertInterval.cs
public class InsertInterval
{
    public int[][] Insert(int[][] intervals, int[] newInterval)
    {
        List<int[]> result = new List<int[]>();
        bool isImported = false;

        if (intervals == null || intervals.Length == 0)
        {
            result.Add(newInterval);
            return result.ToArray();
        }
        
        for (int i = 0; i < intervals.Length; i++)
        {
            if (intervals[i][1] < newInterval[0])
            {
                result.Add(intervals[i]);
            }
            else if (intervals[i][0] <= newInterval[1])
            {
                newInterval[0] = Math.Min(newInterval[0], intervals[i][0]);
                newInterval[1] = Math.Max(newInterval[1], intervals[i][1]);
            }
            else
            {
                if (!isImported)
                {
                    result.Add(newInterval);
                    isImported = true;
                }
                
                result.Add(intervals[i]);
                
            }
        }
        if (!isImported)
        {
            result.Add(newInterval);
            isImported = true;
        }
        return result.ToArray();
    }    
}

File: src/LeetCodePY/insertInterval.py
from typing import List

class Solution:
    def insert(self, intervals: List[List[int]], newInterval: List[int]) -> List[List[int]]:
        if not intervals:
            return [newInterval]

        result : List[List[int]] = []

        for interval in intervals:
            if interval[1] < newInterval[0]:
                result.append(interval)
            elif interval[0] > newInterval[1]:
                result.append(newInterval)
                newInterval = interval
            else:
                newInterval[0] = min(newInterval[0], interval[0])
                newInterval[1] = max(newInterval[1], interval[1])

        result.append(newInterval)
        return result